samcli/vendor/serverlessrepo/exceptions.py

```diff
@@ -6,9 +6,10 @@ class ServerlessRepoError(Exception):
 
     MESSAGE = ""
 
-    def __init__(self, **kwargs):
+    def __init__(self, message=None, **kwargs):
         """Init the exception object."""
-        Exception.__init__(self, self.MESSAGE.format(**kwargs))
+        message = self.MESSAGE.format(**kwargs) if message is None else message
+        Exception.__init__(self, message)
```

Comment:
This is to allow publish exceptions to be created with dynamic messages instead of hardcoded in the class